File: javascript/test/regress_simple.js
#!/usr/bin/env node
const liba = require("../liba");
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
function do_leak_check() {
    for (const arg of arguments) {
        arg.delete();
    }
    if (liba.do_leak_check) {
        liba.do_leak_check();
    }
}
var x = [0, 1, 2, 3, 4];
var y = [1, 2, 3, 4, 5];
var ctx = new liba.regress_simple();
ctx.ols_(x, y, 2, 3);
console.log(ctx.evar(y[0]), ctx.eval(x[0]), ctx.coef, ctx.bias);
ctx.zero();
ctx.olsx(x, y, 2);
console.log(ctx.evar(y[0]), ctx.eval(x[0]), ctx.coef, ctx.bias);
ctx.zero();
ctx.olsy(x, y, 3);
console.log(ctx.evar(y[0]), ctx.eval(x[0]), ctx.coef, ctx.bias);
ctx.zero();
ctx.ols(x, y);
console.log(ctx.evar(y[0]), ctx.eval(x[0]), ctx.coef, ctx.bias);
ctx.zero();
assert("ctx.coef != undefined");
assert("ctx.bias != undefined");
ctx.coef = ctx.coef;
ctx.bias = ctx.bias;
do_leak_check(ctx);

File: java/test/_regress_simple.java
import liba.regress_simple;

class _regress_simple {
    public static void main(String[] args) {
        double x[] = {0, 1, 2, 3, 4};
        double y[] = {1, 2, 3, 4, 5};
        regress_simple obj = new regress_simple(1, 0);
        obj.set_coef(obj.coef());
        obj.set_bias(obj.bias());
        obj.ols_(x, y, 2, 3);
        for (int i = 0; i < x.length; ++i) {
            System.out.format("%g,%g\n", obj.evar(y[i]), obj.eval(x[i]));
        }
        obj.zero();
        obj.olsx(x, y, 2);
        for (int i = 0; i < x.length; ++i) {
            System.out.format("%g,%g\n", obj.evar(y[i]), obj.eval(x[i]));
        }
        obj.zero();
        obj.olsy(x, y, 3);
        for (int i = 0; i < x.length; ++i) {
            System.out.format("%g,%g\n", obj.evar(y[i]), obj.eval(x[i]));
        }
        obj.zero();
        obj.ols(x, y);
        for (int i = 0; i < x.length; ++i) {
            System.out.format("%g,%g\n", obj.evar(y[i]), obj.eval(x[i]));
        }
        obj.zero();
    }
}
